R/model.R

```diff
@@ -239,18 +264,22 @@ compile_method <- function(quiet = TRUE,
     include_paths <- paste0("STANCFLAGS += --include_paths=", include_paths)
   }
 
-  exe <- cmdstan_ext(exe) # adds .exe on Windows
+  # TODO(Rok): Once we handle stancflags separately this should be overriden
+  # if a user specifies their own name
+  model_name_stancflag <- paste0("STANCFLAGS+=--name=", sub(" ", "_", model_name))
+
   run_log <- processx::run(
     command = make_cmd(),
-    args = c(exe, include_paths),
+    args = c(tmp_exe, include_paths, model_name_stancflag),
     wd = cmdstan_path(),
     echo_cmd = !quiet,
     echo = !quiet,
     spinner = quiet,
     stderr_line_callback = function(x,p) { if(quiet) message(x) },
     error_on_status = TRUE
   )
-
+  
+  file.copy(tmp_exe, exe)
```

Comment:
I think this should be

```
file.copy(tmp_exe, exe, overwrite = TRUE)
```

otherwise the outdated binary never gets overwritten.

Reply:
Good call.